src/features/comments/commentsSlice.ts

```diff
@@ -0,0 +1,99 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Comment } from '../../types/Comment';
+import {
+  createComment, deleteComment, getPostComments,
+} from '../../api/comments';
+
+type Comments = {
+  comments: Comment[],
+  isLoading: boolean,
+  hasError: boolean,
+};
+
+const initialState: Comments = {
+  comments: [],
+  isLoading: false,
+  hasError: false,
+};
+
+export const fetchComments = createAsyncThunk(
+  'comments/fetch', (id: number) => {
+    return getPostComments(id);
+  },
+);
+
+export const addComment = createAsyncThunk(
+  'comments/add', (data: Omit<Comment, 'id'>) => {
+    return createComment(data);
+  },
+);
+
+export const delComment = createAsyncThunk(
+  'comments/delete', (id: number) => {
+    return deleteComment(id);
+  },
+);
+
+const postSlice = createSlice({
```

Comment:
rename it to `commentsSlice`

also, according to task requirements, you should also create `selectedPostSlice`:

![image](https://github.com/mate-academy/react_redux-list-of-posts/assets/86923089/0066b1db-288d-465b-a61b-3109da593191)